packages/client/src/hooks/useAccount.js

```diff
@@ -0,0 +1,61 @@
+import { mutate, query, tx } from "@onflow/fcl";
+import { GetAccountBalanceByContractName, SEND_TOKENS_TO_TREASURY } from "flow";
+import { COIN_TYPE_TO_META } from "constants/maps";
+import { COIN_TYPES } from "constants/enums";
+import { REGULAR_LIMIT } from "constants/constants";
+
+const COIN_TYPE_LIST = [COIN_TYPES.FLOW, COIN_TYPES.FUSD, COIN_TYPES.USDC];
+
+export default function useAccount() {
+  const doSendTokensToTreasury = async (treasuryAddr, amount, coinType) => {
+    const vaultPath = COIN_TYPE_TO_META[coinType].storageVaultPath;
+    return await mutate({
+      cadence: SEND_TOKENS_TO_TREASURY,
+      args: (arg, t) => [arg(treasuryAddr, t.Address), arg(amount, t.UFix64), arg(vaultPath, t.Path)],
+      limit: REGULAR_LIMIT,
+    });
+  };
+
+  //FOR TESTING UTILS ONLY, calling this method deposits tokens to treasury
+  //should only be used for testing
+  const initDepositTokensToTreasury = async (treasuryAddr) => {
+    for await (const coinType of COIN_TYPE_LIST) {
+      try {
+        const res = await doSendTokensToTreasury(treasuryAddr, String(parseFloat(10).toFixed(8)), coinType);
+        await tx(res).onceSealed();
+      } catch (err) {
+        console.log(`Failed to deposit ${coinType}, error: ${err}`);
+      }
+    }
+  };
+
+  const getBalanceByCoinType = async (coinType, address) => {
+    const cadence = GetAccountBalanceByContractName(COIN_TYPE_TO_META[coinType].contractName);
+    const vaultPath = COIN_TYPE_TO_META[coinType].publicBalancePath;
+    const balance = await query({
+      cadence,
+      args: (arg, t) => [arg(address, t.Address), arg(vaultPath, t.Path)],
+    });
+    return balance;
+  };
+
+  const getUserBalances = async (address) => {
+    const result = COIN_TYPE_LIST.map((coinType) => ({
```

Comment:
this function can be reused to get all supported tokens balance for Deposit scenario, can we just use the full list here: https://github.com/DapperCollectives/VESSEL/blob/763c98f1a580704e27c0509870091b3591ee2940/packages/client/src/constants/enums.js#L6-L13

Reply:
BLT, STARLY and REVV are not deployed in emulators right now as far as I know, it will cause errors

Reply:
Can we catch those errors and just logging it? We probably don't have to deploy all contracts on emulator, should be able to test it out on testnet.

Reply:
It will cause errors on staging as well - if the vault of that type is not added - and we try to borrow a reference before it's created. We might use instead the function for fetching treasury identifiers, and then based on the identifiers get the vault balance

Reply:
But this is for the address account not for the treasury

Reply:
oh it's just for testing 🙈 then it's not an issue at all

Reply:
On staging or mainnet it is an issue as well, when user logs in with blocto or any other third party wallet, it's not guaranteed they have a FUSD, USDC, STARLY... vaults set up.  I think we should create a separate script for checking the vault existence, then query for their balance.

Reply:
Yea make sense! 👍 I didn’t think through on that earlier.

Reply:
Since https://linear.app/dappercollectives/issue/VSL-166  is effectively the same issue but only for FUSD, I'm going to change its description and tackle the above issue with this ticket